brave/browser-laptop
# History fixes and misc about page fixes

-----

redo of https://github.com/brave/browser-laptop/pull/5559

**NOTE**: _I thought I had fixed the history issues with this, but I was wrong. I removed the issues I thought I had fixed from this PR_

-----

- History code was refactored
  - one step forward with regard to https://github.com/brave/browser-laptop/issues/3502
  - session helper created for aboutHistory (tests added)
  - history specific methods broke out into historyUtil (tests added)
  - history stores it's data in AppStore.about.history (similar structure to new tab code)

- about:history is only updated when something changed
  - number of calls to render on about:history is drastically reduced (should fix https://github.com/brave/browser-laptop/issues/5382; please reopen if not)

- about:brave text update
Implements followup feedback for https://github.com/brave/browser-laptop/pull/5436#issuecomment-259144537

Auditors: @cezaraugusto, @darkdh, @bbondy 

-----

Remove unnecessary/broken/useless context menu items for about pages

Fixes https://github.com/brave/browser-laptop/issues/4812

-----

Update icon for about pages to fa-list (except for about:newtab)

Fixes https://github.com/brave/browser-laptop/issues/5497

Auditors: @bradleyrichter

Test Plan:
1. Launch Brave and visit about:about
2. Open each of the links and confirm it's the `fa-list` icon

## Screenshots
Here's one showing:
- fixing the casing on component names in about:brave
- new "fa-list" urlbar icon
- context menu has  items removed from about pages
  - save as
   - print
   - view source
   - inspect element

![screen shot 2016-11-11 at 5 13 56 pm](https://cloud.githubusercontent.com/assets/4733304/20233934/55db71e0-a832-11e6-962a-01c23abc3ca9.png)
